webapp/src/components/App/Singlestudy/explore/Modelization/Areas/Storages/StorageConfig/AdditionalConstraints/ConstraintForm/OccurrencesFE/OccurrencesTableDialog/utils.ts

```diff
@@ -0,0 +1,91 @@
+/**
+ * Copyright (c) 2025, RTE (https://www.rte-france.com)
+ *
+ * See AUTHORS.txt
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ *
+ * SPDX-License-Identifier: MPL-2.0
+ *
+ * This file is part of the Antares project.
+ */
+
+import { HOURS_IN } from "@/utils/date/constants";
+import * as R from "ramda";
+import type { ConstraintValues } from "../../utils";
+
+export type RowData = Record<string, boolean>;
+
+export type OccurrencesData = Record<string, RowData>;
+
+export const resizeRow = R.curry((resizeValue: number, rowData: RowData) => {
+  const indexes = R.range(0, resizeValue);
+  const values = R.map((i) => rowData[i] || false, indexes);
+  return R.zipObj(indexes, values);
+});
+
+export const occurrencesToGridData = (occurrences: ConstraintValues["occurrences"]) => {
+  const data: OccurrencesData = {};
+
+  for (let hour = 1; hour <= HOURS_IN.WEEK; hour++) {
+    const rowData: RowData = {};
+
+    occurrences.forEach(({ hours }, index) => {
+      rowData[index] = hours.includes(hour);
+    });
+
+    data[hour] = rowData;
+  }
+
+  return data;
+};
+
+export const gridDataToOccurrences = (data: OccurrencesData): ConstraintValues["occurrences"] => {
+  return Object.entries(data).reduce<ConstraintValues["occurrences"]>(
+    (occurrencesAcc, [hour, rowData]) => {
+      Object.values(rowData).forEach((active, i) => {
+        occurrencesAcc[i] ??= { hours: [] };
+        if (active) {
+          occurrencesAcc[i].hours.push(Number(hour));
+        }
+      });
+
+      return occurrencesAcc;
+    },
+    [],
+  );
+};
```

Comment:
I think `Object.values(rowData)` doesn't guarantee order. Should use `Object.entries()`:

Reply:
For `Object.values()`:

> The order of the array returned by Object.values() is the same as that provided by a [for...in](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...in) loop.

Source: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/values

For `for...in`:

> Within each component of the prototype chain, all non-negative integer keys (those that can be array indices) will be traversed first in ascending order by value, then other string keys in ascending chronological order of property creation.

Source: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...in

`OccurrencesData` uses integer keys.